File: 10539.c
#include <stdio.h>

int main() {
	int N;
	int A[101];
	int B[101];
	int sum;
	int temp;

	scanf("%d", &N);

	for (int i = 1; i <= N; i++) {
		scanf("%d", &B[i]);
	}

	temp = 0;
	for (int i = 1; i <= N; i++) {
		sum = B[i] * i;
		A[i] = sum - temp;
		temp += A[i];
	}

	for (int i = 1; i < N; i++)
		printf("%d ", A[i]);
	printf("%d\n", A[N]);

	return 0;
}

File: 10539.py
N = int(input())
B = map(int,input().split())
A = []
sum = 0
for i in B:
    if len(A)==0:
        A.append(i)
        sum = i
    else:
        A.append(i*(len(A)+1)-sum)
        sum += A[-1]
    #print(i,sum,A[-1])
#print(A)
for i in A:
    print(i, end=" ")
